src/helpers.js

```diff
@@ -319,6 +320,39 @@ export async function processESLintMessages(messages, textEditor, showRule, work
   }))
 }
 
+/**
+ * Update the list of fixable rules
+ */
+function updateFixableRules() {
+  fixableRules.clear()
+  lintRules.forEach((props, rule) => {
+    if (
+      Object.prototype.hasOwnProperty.call(props, 'meta') &&
+      Object.prototype.hasOwnProperty.call(props.meta, 'fixable')
+    ) {
+      fixableRules.add(rule)
+    }
+  })
+}
+
+/**
+ * Process the updated rules into the local Map and call further update functions
+ * @param  {Array} updatedRules Array of Arrays of the rule and properties
+ */
+export function updateRules(updatedRules) {
```

Comment:
__*Preference*__

I'm not sure `update` is the best name here, because it kind of makes it sound like like the old rules are being kept and just somehow modified.  I know this is niggling, but I think I'd prefer a name like `replaceRules` in this case.

Reply:
I went with `update` since as far as anything outside this function is concerned it _is_ an update. The fact that it happens to clear the current contents and then fill in new ones is an implementation detail.

Reply:
I see your reasoning.  Personally, if I call `updateThings(newThing)`, it isn't immediately clear as a consumer of the function whether all the old things will be gone or not.  I might think that I'm just adding a new thing to the existing things.  I guess all I'm saying is that `update` is vague, and I find `replace` to be clearer, but I'm only at a 3 of 10 on the [SSOGAF](http://blog.capwatkins.com/the-sliding-scale-of-giving-a-fuck).

Reply:
It's certainly not ideal that it clears the old rules in place as a side-effect before updating them. But the general concept of full replacement as an update is totally valid, and the foundation of immutability and great tools like Redux.

Reply:
I have no issue with doing it that way, just the _naming_.

Reply:
Updating in place gets complex since then you have to determine what you need to remove, and what you need to insert. Since there is nothing actually _tied_ to the contents here a simple replacement works just fine.

I'd say I'm at around a 4 or 5 on the SSOGAF 😛.

Reply:
Haha, you called it a _replace_ment.  😝